internal/services/topic.go

```diff
@@ -183,6 +191,36 @@ func (ts *TopicService) assignments(currentPartitions int, brokersNumber int) (m
 	return assignments, int(minISR)
 }
 
+func (ts *TopicService) alterAssignments(topic string, assignments [][]int32) error {
```

Comment:
Why make it take the topic as a parameter when it's available as `ts.canaryConfig.Topic`?

Also a line of doc about what this function is trying to achieve wouldn't hurt.